flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/RexNodeExtractor.scala

```diff
@@ -335,7 +336,7 @@ class RexNodeToExpressionConverter(
 
       case TIMESTAMP_WITHOUT_TIME_ZONE =>
         val v = literal.getValueAs(classOf[java.lang.Long])
```

Comment:
Why the literal for `TIMESTAMP_WITHOUT_TIME_ZONE` is `long`? I remember somewhere else you presume the literal is `TimestampString`

Reply:
Good catch. It should be TimestampString and preserve the nanosecond precision.